lib/ClangImporter/ClangImporter.cpp

```diff
@@ -2827,6 +2827,12 @@ void ClangModuleUnit::collectLinkLibraries(
     return;
   for (auto clangLinkLib : clangModule->LinkLibraries) {
     LibraryKind kind;
+
+    // Skip this lib name in favor of export_as name.
+    if (clangModule->Name == clangLinkLib.Library &&
```

Comment:
Same comment as with the Clang PR; I don't think we should be checking `clangModule->Name == clangLinkLib.Library` here, because the re-exporting module map should redeclare whatever is needed.